src/mscorlib/shared/System/MemoryExtensions.cs

```diff
@@ -707,21 +707,82 @@ public static bool EndsWith<T>(this ReadOnlySpan<T> span, ReadOnlySpan<T> value)
         }
 
         /// <summary>
-        /// Creates a new span over the portion of the target array.
+        /// Creates a new span over the target array.
         /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Span<T> AsSpan<T>(this T[] array)
         {
             return new Span<T>(array);
         }
 
+        /// <summary>
+        /// Creates a new Span over the portion of the target array beginning
+        /// at 'start' index and ending at 'end' index (exclusive).
+        /// </summary>
+        /// <param name="array">The target array.</param>
+        /// <param name="start">The index at which to begin the Span.</param>
+        /// <param name="length">The number of items in the Span.</param>
+        /// <remarks>Returns default when <paramref name="array"/> is null.</remarks>
+        /// <exception cref="System.ArrayTypeMismatchException">Thrown when <paramref name="array"/> is covariant and array's type is not exactly T[].</exception>
+        /// <exception cref="System.ArgumentOutOfRangeException">
+        /// Thrown when the specified <paramref name="start"/> or end index is not in the range (&lt;0 or &gt;=Length).
+        /// </exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Span<T> AsSpan<T>(this T[] array, int start, int length)
```

Comment:
Where is the overload that takes just index?

Reply:
That one can't be done portably. It already exists in `MemoryExtensions.Fast.cs`.

Reply:
> That one can't be done portably.

What do you mean by that? We would need a portable specific implementation, but why can't it be done?

Reply:
We don't have a constructor that takes just an array and offset. So we have a implemention in MemoryExtensions.Fast.cs for netcore (this was done in the prior PR) and we'll need to have a separate netstandard implemention in corefx. But we're not adding the overload in this PR because it already exists in the Fast file.